src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory_Methods.cs

```diff
@@ -31,11 +29,9 @@ private static ImmutableArray<IOperation> ToStatements(BoundStatement statement)
             return ImmutableArray.Create(Create(statement));
         }
 
-        private static readonly ConditionalWeakTable<BoundIncrementOperator, ILiteralExpression> s_incrementValueMappings = new ConditionalWeakTable<BoundIncrementOperator, ILiteralExpression>();
-
-        private static ILiteralExpression CreateIncrementOneLiteralExpression(BoundIncrementOperator boundIncrementOperator)
+        private ILiteralExpression CreateIncrementOneLiteralExpression(BoundIncrementOperator boundIncrementOperator)
         {
-            return s_incrementValueMappings.GetValue(boundIncrementOperator, (increment) =>
+            return _cache.GetOrCreateOperationFrom(boundIncrementOperator, nameof(CreateIncrementOneLiteralExpression), (increment) =>
```

Comment:
>_cache.GetOrCreateOperationFrom(boundIncrementOperator, nameof(CreateIncrementOneLiteralExpression), (increment) [](start = 19, length = 112)

Do we ever call this function more than once for the same bound node? It looks like there is no real need to cache this node in the ```_cache```, this function could instead return an instance of ```Lazy<IOperation>```, which will calculate the value lazily and cache the result afterwards.